sdk/rust/oak_runtime/src/lib.rs

```diff
@@ -29,6 +29,8 @@ use std::sync::{Arc, RwLock};
 
 pub mod proto;
 
+pub type Handle = u64;
```

Comment:
Could this be made to use the one defined in the SDK? I think it would be fine (in fact, desirable) to have the runtime depend on the SDK crate (not the other way round of course).

Reply:
I think this is backwards- `sdk/oak` is the safe interface that wraps WASM imported functions. `sdk/oak_runtime` is providing the WASM functions and so follows the ABI definitions (e.g. handle==u64).

I think the intention is for `oak_runtime` to move out of the `sdk` directory eventually, but currently it only powers the tests. @daviddrysdale does that sound right?

Reply:
Yeah, that sounds about right.  

It would be nice to share definitions though, so it might make sense to have a little `oak_abi` crate that both the (Rust) runtime and the `oak` SDK can depend on. Looking at the current `oak_runtime/` code, I think it would currently just have:

 - the `oak::Handle` typedef
 - the `OakStatus` & `OakChannelStatus` enums (from protobuf definitions)
 - the `oak::wasm::INVALID_HANDLE` constant.

Reply:
That's what I meant, except I think it's fine for the runtime crate to depend on the SDK crate; anyways it's fine to add a shared `oak_abi` crate that both depends on if you prefer.

Reply:
The runtime crate depending on the SDK crate seems like a layering inversion &ndash; it should theoretically be possible for a masochistic Node author to write raw Wasm code and therefore never even install the SDK crate(s).

Reply:
Not sure I follow; I am just saying that we can consider the sdk crate as the "public interface" of the runtime crate, so it seems fine to have the runtime depend on the sdk crate, and have all "public" types in the sdk crate but also be able to use them from the runtime crate internally. I don't see any layering violation there; no one externally would ever use the runtime crate directly, right?

Or, again, as @daviddrysdale suggested, it's also fine to have a shared crate that both sdk and runtime depend on.

Reply:
I don't think the sdk crate is the public interface of the runtime &ndash; the collection of host functions that make up the ABI is the public interface of the runtime.  The SDK is then a collection of libraries that make it (much) easier for Rust code to use that ABI; if a Node were written in (say) C++ then that SDK wouldn't be available.

To put it in analogy terms with UNIX:
 - Oak runtime ≈ UNIX kernel
 - Oak host functions ABI ≈ UNIX system call interface
 - Oak SDK ≈ libc (and hopefully other user-space libraries too).

It would be odd for the Linux kernel to depend on a header file from glibc; if nothing else, it would prevent the kernel being used with (say) [musl](https://www.musl-libc.org/).

Reply:
Ah got it, thanks for clarifying, I also re-read your previous message more carefully and that does make sense.

Yes, if we want to make the sdk crate optional for node authors (at least in principle), then a minimal ABI crate is the way to go. My point was just that technically it would be fine for the runtime crate to depend on the sdk crate, but not the other way round. Having the ABI one is even better for layering / separation purposes, I agree.

Reply:
#447 is working towards having the Rust runtime not under SDK, and depending on an `oak_abi` crate for ABI type/constant definitions.

Reply:
So is the plan to remove this in favour of the upcoming shared definition? If so, perhaps add a `TODO(#447)` for the time being?

Or are you suggesting keeping both `oak::Handle` (from the SDK) and `oak_abi::Handle` around eventually?